src/main/java/pl/team/carrent/promotion/PromotionRepository.java

```diff
@@ -49,9 +52,26 @@
             " where UPPER(cm.segment) = UPPER(:segment)", nativeQuery = true)
     List<Promotion> findPromotionByCarSegment(String segment);
 
-    @Query(value = "select p.* from promotion p\n" +
-            "left join promotion_cars pca on p.id=pca.promotion_id\n" +
-            "left join promotion_clients pci on p.id=pci.promotion_id\n" +
-            "where (pca.cars_id=:carId or pci.clients_id=:clientId) and (:date between p.promo_start and p.promo_end);", nativeQuery = true)
+    @Query(value = "select p.* from promotion p" +
+            " join promotion_clients pcli on pcli.promotion_id=p.id" +
+            " join client c on c.id=pcli.clients_id " +
+            " where UPPER(c.name) = UPPER(:name)", nativeQuery = true)
+    List<Promotion> findPromotionByClientName(String name);
+
+    @Query(value = "select p.* from promotion p" +
+            " left join promotion_cars pca on p.id=pca.promotion_id" +
+            " left join promotion_clients pcli on p.id=pcli.promotion_id" +
+            " where (pca.cars_id=:carId or pcli.clients_id=:clientId) and (:date between p.promo_start and p.promo_end);", nativeQuery = true)
     List<Promotion> findAllMatchedPromotions(LocalDate date, int carId, int clientId);
+
+    @Query(value = "select p.* from promotion p" +
+            " left join promotion_cars pca on p.id=pca.promotion_id" +
+            " where (pca.cars_id=:carId) and (:date between p.promo_start and p.promo_end);", nativeQuery = true)
+    List<Promotion> findAllMatchedPromotionsByCarId(LocalDate date, int carId);
+
+    @Query(value = "select p.* from promotion p" +
+            " left join promotion_clients pcli on p.id=pcli.promotion_id" +
+            " where (pcli.clients_id=:clientId) and (:date between p.promo_start and p.promo_end);", nativeQuery = true)
+    List<Promotion> findAllMatchedPromotionsByClientId(LocalDate date, int clientId);
```

Comment:
i tutaj też można view

Reply:
Co do tej kosmetyki, to kolejny etap wtajemniczenia do przegadania przy piwie;) póki co satysfakcjonuje mnie, że działa:)